indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -300,4 +338,50 @@ private String defaultDir(@Nullable String configParameter, final String default
 
     return configParameter;
   }
+
+  private static class TaskStorageDirTracker
```

Comment:
This class doesn't belong in a Config class.  A Config class should be a config class and should not have business logic inside of it.  We can see the reasons for this in that this Config class is already doing some business logic in figuring out the various directories with the methods

```
  public File getTaskDir(String taskId)
  {
    return new File(baseTaskDir, IdUtils.validateId("task ID", taskId));
  }

  public File getTaskWorkDir(String taskId)
  {
    return new File(getTaskDir(taskId), "work");
  }

  public File getTaskTempDir(String taskId)
  {
    return new File(getTaskDir(taskId), "temp");
  }

  public File getTaskLockFile(String taskId)
  {
    return new File(getTaskDir(taskId), "lock");
  }
```

If this Config class didn't have that business logic built into it, it would be a lot simpler to actually extract the `StorageDirTracker`.  By adding more of this complexity here, we are going to be continuing that sin, where maybe we should pay it back down instead?

Reply:
Done